### ISSUE TITLE
🐛 Bug Report: docs building backends / migrating / auth

### ISSUE THREAD

**Issue by richardjq** [backstage/backstage#24018]
### 📜 Description

In `docs/backend-system/building-backends/08-migrating.md`, under "Using Legacy Providers" there is a code block with the line:

```
backend.add(legacyPlugin('auth'), import('./plugins/auth'));
```

The parens are incorrect. 

### 👍 Expected behavior

Syntax should be:

```
backend.add(legacyPlugin('auth', import('./plugins/auth')));
```

### 👎 Actual Behavior with Screenshots

Using as is, you will get the below error message:

```
$ tsc
packages/backend/src/index.ts:16:13 - error TS2554: Expected 2 arguments, but got 1.

16 backend.add(legacyPlugin('auth'), import('./plugins/auth'));
               ~~~~~~~~~~~~~~~~~~~~

  node_modules/@backstage/backend-common/dist/index.d.ts:138:44
    138 declare const legacyPlugin: (name: string, createRouterImport: Promise<{
                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    139     default: LegacyCreateRouter<TransformedEnv<{
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    ... 
    153     }>>;
        ~~~~~~~~
    154 }>) => _backstage_backend_plugin_api.BackendFeature;
        ~~
    An argument for 'createRouterImport' was not provided.

packages/backend/src/index.ts:16:35 - error TS2554: Expected 1 arguments, but got 2.

16 backend.add(legacyPlugin('auth'), import('./plugins/auth'));
                                     ~~~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: packages/backend/src/index.ts:16

error Command failed with exit code 2.
```

### 👟 Reproduction steps

Implement the example as provided and build backstage.

### 📃 Provide the context for the Bug.

I am working on updating to the new backend using an OIDC Auth plugin. 

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by tudi2d** [comment]
Thanks for reporting this! Feel free to raise a PR to fix this in the documentation :)

**Comment by pjungermann** [comment]
This was resolved by PR #24100
